src/bio.c

```diff
@@ -725,6 +725,28 @@ long wolfSSL_BIO_ctrl(WOLFSSL_BIO *bio, int cmd, long larg, void *parg)
     }
     return ret;
 }
+
+/* Increment the WOLFSSL_BIO ref count by one, prevents BIO from being
+ * freed until ref count is back down to 1.
+ *
+ * bio  the structure to increment ref count
+ *
+ * returns 1 on success, 0 on failure */
+
+int wolfSSL_BIO_up_ref(WOLFSSL_BIO* bio)
+{
+    if (bio) {
+        if (wc_LockMutex(&bio->refMutex) != 0) {
```

Comment:
In other places in the code, I've seen the locking/unlocking guarded by `#ifndef SINGLE_THREADED`. May want to add that here, too.

Reply:
I'm not sure an explicit ifndef is needed since when SINGLE_THREADED is defined, wc_LockMutex() and others are still available.  They just return 0, in wc_port.c.

Reply:
I agree but we have moved to guarding all uses of the mutex with `SINGLE_THREADED`. Please let's be consistent.

Reply:
Added SINGLE_THREADED protections